runtime/onert/core/src/exporter/train/CheckpointExporter.cc

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "exporter/train/CheckpointExporter.h"
+
+#include "exec/Execution.h"
+#include "ir/train/TrainingInfo.h"
+#include "ir/train/Checkpoint.h"
+
+#include <fstream>
+#include <iostream>
+#include <numeric>
+
+namespace
+{
+
+using namespace onert;
+using namespace ir;
+using namespace train;
+using namespace checkpoint;
+using namespace exec;
+
+class CheckpointExporter
+{
+public:
+  CheckpointExporter(const std::unique_ptr<ir::train::TrainingInfo> &train_info,
+                     const std::unique_ptr<exec::Execution> &exec)
```

Comment:
(Optional)
If you would like to keep constness of those `unique_ptr`s it would be better to pass by const raw pointer instead const reference of `unique_ptr`.

Reply:
@ragmani 

Could you explain more about your suggestion? I hardly catch the intention of your suggestion. :)

Reply:
`const uniquie_ptr`'s data can be modified within the function. `const uniquie_ptr` has constness of pointer. For example,
```c++
  CheckpointExporter(const std::unique_ptr<ir::train::TrainingInfo> &train_info,
                     const std::unique_ptr<exec::Execution> &exec)
  {
    train_info->setVersion(1); // no error
    train_info = nullptr // compile error
  }
```

But `const raw pointer's data cannot be modified within the function. For example,
```c++
  CheckpointExporter(const ir::train::TrainingInfo *train_info,
                     const exec::Execution *exec)
  {
    train_info->setVersion(1); // compile error
    train_info = nullptr // no error, but no problem outside the function if the pointer address is modified
  }
```

In other words, we cannot keep data constness of parameters passing by const reference of `unique_ptr`. So, the parameter's data may be modified by accident.

Reply:
@ragmani 

I applied your suggestion. PTAL again :)